src/hats_import/catalog/run_import.py

```diff
@@ -148,3 +154,23 @@ def run(args, client):
             step_progress.update(1)
             assert is_valid_catalog(args.catalog_path)
             step_progress.update(1)
+
+
+def create_data_thumbnail(args, client, resume_plan):
+    """Create a thumbnail of the data in the catalog, top row from each partition."""
+    output_catalog_path = get_upath(args.catalog_path)
+    dataset_subdir = output_catalog_path / "dataset"
+    _, dataset = io.read_parquet_dataset(
+        dataset_subdir,
+        ignore_prefixes=["intermediate", "_common_metadata", "_metadata"],
+        exclude_invalid_files=True,
+    )
+    futures = [
+        client.submit(lambda pq_path: pd.read_parquet(pq_path, engine="pyarrow").head(1), p)
+        for p in dataset.files
```

Comment:
What if there are 1_000_000 partitions, but our threshold is only 10_000? Can we clip the number of futures generated to the max number of rows?

Reply:
This is `args.pixel_threshold`.  What I will do is randomly sample which partitions to take the top row from, if there are more partitions than this threshold.